Sources/EpoxyCollectionView/CollectionView/Internal/CollectionViewScrollToItemHelper.swift

```diff
@@ -333,70 +318,77 @@ final class CollectionViewScrollToItemHelper {
       forTargetItemIndexPath: targetIndexPath,
       collectionView: collectionView)
 
+    let insetBounds = collectionView.bounds.inset(by: collectionView.adjustedContentInset)
+
     switch (scrollAxis, positionRelativeToVisibleBounds) {
     case (.vertical, .before):
       return .top
     case (.vertical, .after):
       return .bottom
     case (.vertical, .partiallyOrFullyVisible(let itemFrame)):
-      guard !collectionView.bounds.contains(itemFrame) else { return nil }
-      return itemFrame.midY < collectionView.bounds.midY ? .top : .bottom
+      guard !insetBounds.contains(itemFrame) else { return nil }
+      return itemFrame.midY < insetBounds.midY ? .top : .bottom
     case (.horizontal, .before):
       return .left
     case (.horizontal, .after):
       return .right
     case (.horizontal, .partiallyOrFullyVisible(let itemFrame)):
-      guard !collectionView.bounds.contains(itemFrame) else { return nil }
-      return itemFrame.midX < collectionView.bounds.midX ? .left : .right
+      guard !insetBounds.contains(itemFrame) else { return nil }
+      return itemFrame.midX < insetBounds.midX ? .left : .right
     default:
       EpoxyLogger.shared.assertionFailure("Unsupported scroll position.")
       return nil
     }
   }
 
-  // Returns the correct resting position for a scroll-to-item action for the current viewport.
-  // This will be used to determine how much farther we need to programmatically scroll on each
-  // animation tick.
-  private func targetOriginForVisibleItem(
+  /// Returns the correct content offset for a scroll-to-item action for the current viewport.
+  ///
+  /// This will be used to determine how much farther we need to programmatically scroll on each
+  /// animation tick.
+  private func targetContentOffsetForVisibleItem(
```

Comment:
I reworked this to return content offsets instead of item-origin relative distances so that we could clamp it to the min/max content offset of the scroll view